src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -196,10 +198,41 @@ public void deleteAttachment(String requestId, String companyNumber, String obje
                 () -> new AttachmentNotFoundException(String.format(ATTACHMENT_NOT_FOUND_MESSAGE, attachmentId))
         );
 
-        attachments.remove(attachment);
+        boolean successfulDelete = deleteFromS3(requestId, companyNumber, objectionId, attachmentId);
 
-        objection.setAttachments(attachments);
+        if(successfulDelete) {
+            attachments.remove(attachment);
 
-        objectionRepository.save(objection);
+            objection.setAttachments(attachments);
+
+            objectionRepository.save(objection);
+        }
+    }
+
+    private boolean deleteFromS3(String requestId, String companyNumber, String objectionId, String attachmentId) {
```

Comment:
Looks like companyNumber and objectionId aren't needed/used in this method. I think these parameters can just be ommitted.

Reply:
They are needed for the log map

Reply:
But all we do is log them. They aren't actually used or needed by the method.

Reply:
Moved it to a separate method

Reply:
Sorry, but that's not addressing my comment - the 'deleteFromS3' method doesn't require a company number or objection Id in order to do it's job. Therefore these things shouldn't be passed in. If we think that these values should be written to the log then maybe that should be job of the client method that calls 'deleteFromS3'.

Reply:
To be honest I'm not sure what you have in mind.